src/QsCompiler/Tests.Compiler/TestCases/PopulateCallGraph.qs

```diff
@@ -134,6 +134,81 @@ namespace Microsoft.Quantum.Testing.PopulateCallGraph {
 
 // =================================
 
+// Multiple Entry Points
+namespace Microsoft.Quantum.Testing.PopulateCallGraph {
+
+    @EntryPoint()
+    operation Main1() : Unit {
+        Foo();
+    }
+
+    @EntryPoint()
+    operation Main2() : Unit {
+        Bar();
+    }
+
+    operation Foo() : Unit { }
+
+    operation Bar() : Unit { }
+}
+
+// =================================
+
+// Unit Test as Starting Point
+namespace Microsoft.Quantum.Testing.PopulateCallGraph {
+
+    open Microsoft.Quantum.Diagnostics;
+
+    @Test("QuantumSimulator")
+    operation Test() : Unit {
+        Foo();
+    }
+
+    operation Foo() : Unit { }
+
+    operation Bar() : Unit { }
```

Comment:
(Minor)
It is somewhat suspicious to see multiple _identical_ definitions (for `Foo()`, `Bar()`, `Main1()`, `Main2()`) in the same namespace. It causes some redundant questions, e.g. "What if one of the two definitions of `Foo()` is updated, but the other is not? Will that cause a compiler error? If yes, then will that error be clear?".

What if in green lines 167 - 169 we remove the definitions for `Foo()` and `Bar()` (assuming that we reuse those from lines 150 - 152)?
Same for `Main1()` and `Main2()`.

Reply:
The testing logic breaks up the unit tests into totally separate compilations. The edges of the 'chunks' are the lines with the `// =================================`. So the code from lines 150 - 152 are not present in the compilation when we run the test case starting at line 158.

Reply:
To answer the questions you posted, having multiple callables with the same name in the same namespace will cause an error in Q#. As I said this is not an issue here because the different 'chunks' are compiled completely separate from each other.